gwnr/waveform/enigma_utils.py

```diff
@@ -171,33 +171,60 @@ def eccentricity_at_reference_frequency(
     return e0
 
 
-def get_imr_enigma_modes(
+def get_inspiral_enigma_modes(
     mass1,
     mass2,
-    spin1z,
-    spin2z,
-    eccentricity,
-    mean_anomaly,
-    distance,
     f_lower,
     sample_rate,
+    spin1z=0.0,
+    spin2z=0.0,
+    eccentricity=0.0,
+    mean_anomaly=0.0,
+    distance=1.0,
     modes_to_use=[(2, 2), (3, 3), (4, 4)],
     include_conjugate_modes=True,
-    f_mr_transition=None,
-    f_window_mr_transition=None,
-    return_hybridization_info=False,
+    return_orbital_params=False,
     verbose=False,
 ):
-    if f_mr_transition is None:
-        f_mr_transition = 6.0**-1.5 / (mass1 + mass2) / lal.MTSUN_SI / lal.PI
-        # get_isco_frequency(m1, m2, s1z, s2z) / 2 / 2,
-    if f_window_mr_transition is None:
-        f_window_mr_transition = 10.0
-
-    if distance < lal.PC_SI:
-        distance = distance * lal.PC_SI
-
-    itime = time.time()
+     """
+    Returns inspiral ENIGMA GW modes
+    
+    Parameters:
+    -----------
+        mass1, mass2            -- Binary's component masses (in solar masses)
+        f_lower                 -- Starting frequency of the waveform (in Hz)
+        sample_rate             -- Sampling rate of the waveform timeseries (in Hz)
+        spin1z, spin2z          -- z-components of component dimensionless spins (lies in [0,1))
+        eccentricity            -- Initial eccentricity
+        mean_anomaly            -- Mean anomaly of the periastron (in rad).
+        distance                -- Luminosity distance to the binary (in Mpc)
+        modes_to_use            -- GW modes to use. List of tuples (l, |m|)
+        include_conjugate_modes -- If True, (l, -|m|) modes are included as well
+        return_orbital_params   -- If True, returns the orbital evolution of all the orbital elements.
+                                   Can also be a list of orbital variable names to return only those
+                                   specific variables. Available orbital variables are: 
+                                   ['x', 'e', 'l', 'phi', 'phidot', 'r', 'rdot']
+        verbose                 -- Verbosity flag
+    
+    Returns:
+    --------
+        t                 -- Time grid (in seconds)
+        orbital_var_dict  -- Dictionary of evolution of orbital elements. 
+                             Returned only if "return_orbital_params" is specified.
+        modes_numpy       -- Dictionary of GW modes
+    """
+
+    if return_orbital_params:
+        orbital_var_names = ["x", "e", "l", "phi", "phidot", "r", "rdot"]
+        if return_orbital_params != True:
+            for name in return_orbital_params:
+                if name not in orbital_var_names:
+                    raise Exception(f"{name} is not a valid orbital variable name")
```

Comment:
Make this error message a bit more helpful, by adding what the allowed names are for orbital variables.

Reply:
I've added them in the docstring, but I'll add them here also 👍

Reply:
Yes! When this code is run as part of a bigger script, the user should be able to see the error and understand what went wrong. User won't read the docstring of every function called by a big script that they are running.